### PR TITLE
Read vector from disk

### PR DESCRIPTION
**Pull request by godchen0212**
Read vector from disk.
Issue：#6706 

Signed-off-by: godchen <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 1]
```diff
@@ -98,13 +98,25 @@ func (ib *insertBuffer) size(segmentID UniqueID) int32 {
 	var maxSize int32 = 0
```
Comment (cydrain) [inline internal/datanode/flow_graph_insert_buffer_node.go:98]: why not change "maxSize" to int64 type ?

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 2]
```diff
@@ -98,13 +98,25 @@ func (ib *insertBuffer) size(segmentID UniqueID) int32 {
...
 		fdata, ok := data.(*storage.FloatVectorFieldData)
-		if ok && int32(fdata.NumRows) > maxSize {
-			maxSize = int32(fdata.NumRows)
+		totalNumRows := int64(0)
+		if fdata != nil && fdata.NumRows != nil {
+			for _, numRow := range fdata.NumRows {
+				totalNumRows += numRow
+			}
+		}
+		if ok && int32(totalNumRows) > maxSize {
```
Comment (cydrain) [inline internal/datanode/flow_graph_insert_buffer_node.go:107]: this part should be embraced by "if fdata != nil ..."

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 3]
```diff
@@ -98,13 +98,25 @@ func (ib *insertBuffer) size(segmentID UniqueID) int32 {
...
 		bdata, ok := data.(*storage.BinaryVectorFieldData)
-		if ok && int32(bdata.NumRows) > maxSize {
-			maxSize = int32(bdata.NumRows)
+		totalNumRows = int64(0)
+		if bdata != nil && bdata.NumRows != nil {
+			for _, numRow := range bdata.NumRows {
+				totalNumRows += numRow
+			}
+		}
+		if ok && int32(totalNumRows) > maxSize {
```
Comment (cydrain) [inline internal/datanode/flow_graph_insert_buffer_node.go:118]: this part should be embraced by "if bdata != nil ..."

File: internal/datanode/flow_graph_insert_buffer_node.go (hunk 1 above)
Comment (godchen0212) [reply internal/datanode/flow_graph_insert_buffer_node.go:98]: @XuanYang-cn Can I change the type of maxSize to int64？

File: internal/querynode/query_collection.go [hunk 4]
```diff
@@ -63,6 +64,9 @@ type queryCollection struct {
 
 	queryMsgStream       msgstream.MsgStream
 	queryResultMsgStream msgstream.MsgStream
+
+	schemaHelper *typeutil.SchemaHelper
+	vcm          storage.ChunkManager
```
Comment (cydrain) [inline internal/querynode/query_collection.go:69]: suggest keep the naming style as above, use "chunkManager"

Comment (godchen0212) [reply internal/querynode/query_collection.go]: Have changed the type to *storage.VectorChunkManager, the name "vcm" is already clear.

File: internal/querynode/query_collection.go [hunk 5]
```diff
@@ -1059,55 +1068,75 @@ func (q *queryCollection) search(msg queryMsg) error {
...
+				return err
+			}
+			dataType := schema.DataType
+			log.Debug("FillVectorFieldData", zap.Any("datatype", resultFieldData.Type))
+
+			switch dataType {
+			case schemapb.DataType_BinaryVector:
+				x := resultFieldData.GetVectors().GetData().(*schemapb.VectorField_BinaryVector)
+				content := make([]byte, dim)
+				_, err := q.vcm.ReadAt(vecPath, content, offset*dim)
```
Comment (cydrain) [inline internal/querynode/query_collection.go:1115]: should be "q.vcm.ReadAt(vecPath, content, offset*dim/8)"

File: internal/storage/vector_chunk_manager.go [hunk 6]
```diff
@@ -62,13 +56,20 @@ func (vcm *VectorChunkManager) Load(key string) (string, error) {
 			floatData := floatVector.Data
 			result := make([]byte, 0)
 			for _, singleFloat := range floatData {
-				result = append(result, Float32ToByte(singleFloat)...)
+				result = append(result, typeutil.Float32ToByte(singleFloat)...)
```
Comment (cydrain) [inline internal/storage/vector_chunk_manager.go:59]: suggest to use binary.Write

File: internal/storage/data_codec_test.go [hunk 7]
```diff
@@ -142,52 +142,52 @@ func TestInsertCodec(t *testing.T) {
 	insertData1 := &InsertData{
 		Data: map[int64]FieldData{
 			RowIDField: &Int64FieldData{
-				NumRows: 2,
+				NumRows: []int64{2},
```
Comment (xige-16) [inline internal/storage/data_codec_test.go:145]: why change the type of numRows to slice?

Comment (godchen0212) [reply internal/storage/data_codec_test.go:145]: In order to know the size of each file, it is used to determine which file the vector to be searched for is in. 
For example, I know that the offset of the vector in the segment is 700. There are three files in this segment vector column, and the size of each file is 300, 300, 400, 500. I can know that it is in the third file. With numRows of []int64{300,300,400,500}. If I use the previous numRows, I can only know that the segment is 1500 rows in total. But can't know which file the vector I need is in

**Pull request merged**